web/src/lib/components/ui/constraintSignature/constraint-signature.svelte

```diff
@@ -0,0 +1,161 @@
+<script lang="ts">
+  import {
+    type ConstraintSignature,
+    getGrades,
+    getRooms,
+    getStudentGroups,
+    getStudents,
+    getSubjects,
+    getTags,
+    getTeachers,
+    getTimeslots,
+    type Pageable,
+    ParameterType,
+  } from '$lib/sdk/fetch-client';
+  import { Button } from '$lib/elements/ui/button';
+  import ComboBox, { type ComboBoxItem } from '$lib/elements/ui/combo-box/combo-box.svelte';
+
+  export let constraintSignature: ConstraintSignature;
+  export let addInstance: (
+    constraintSignature: ConstraintSignature,
+    data: Record<string, ComboBoxItem[]>,
+  ) => Promise<void>;
+
+  let data: Record<string, ComboBoxItem[]> = {};
+
+  async function updateItems(value: string, name?: string, parameterType?: ParameterType) {
+    const page: Pageable = { page: 0, size: 2 };
+    if (name === undefined) {
+      name = constraintSignature.parameters.find(
+        (parameter) => parameter.parameterType === parameterType!,
+      )!.parameterName!;
+    }
+    try {
+      switch (parameterType) {
+        case ParameterType.Grade: {
+          const grades = await getGrades(page, { name: value });
```

Comment:
Isn't `grades` a page (you pass a page to it)? If so, you should destructure the content, no?